modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc

```diff
@@ -5890,6 +5890,118 @@ Total database accesses: 5, total allocated memory: 1488
 
 ======
 
+[role=label--new-5.21]
+[[query-plan-stateful-shortest-path-into]]
+== StatefulShortestPath(Into)
+// StatefulShortestPath(Into)
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE CONSTRAINT person_name_unique IF NOT EXISTS FOR (p:Person) REQUIRE (p.name) IS UNIQUE
+----
+////
+
+The `StatefulShortestPath(Into)` operator finds shortest paths between two previously matched node variables.
+It uses a bidirectional breadth-first search (BFS) algorithm, which performs two BFS invocations at the same time, one from the left boundary node and one from the right boundary node.
+Once a node is found by both BFS invocations, which indicates that it can be reached from both boundary nodes, the algorithm successfully terminates.
+When both BFS invocations terminate without any overlap in the nodes found, then there exists no valid path between both boundary nodes.
```

Comment:
Do both BFS traversals have to terminate? Or just one? I thought that when one of the BFS exhausts all permitted paths from it's starting node, the whole algorithm terminates. If it didn't do that, wouldn't the disconnected cases perform as poorly as when it was implemented with "unidirectional" BFS?

Reply:
(I moved this comment from the [merged PR](https://github.com/JPryce-Aklundh/docs-cypher/pull/1)